tests/utils/test_backbone_utils.py

```diff
@@ -126,3 +137,51 @@ def test_backbone_mixin(self):
         backbone.out_indices = [-3, -1]
         self.assertEqual(backbone.out_features, ["a", "c"])
         self.assertEqual(backbone.out_indices, [-3, -1])
+
+    @slow
+    @require_torch
+    def test_load_backbone_in_new_model(self):
```

Comment:
The main pain point is `_fast_init` when using `from_pretrained` will not initialize layers. not 100% familiar with the interactions between timm backbone and that but interesting to test `from_pretrained` here as well IMO 😉

Reply:
What behaviour wrt `from_pretrained` do you think should be tested here? It's somewhat separate from the added functionality: a new model with randomly initialized weights can have pretrained backbone weights (which are loaded with `from_pretrained`). We know models that have pretrained weights which contain backbones can be successfully loaded back in, with all the weights intact as this is covered in the model integration tests for e.g. detr and dpt. I could save and reload the models to make sure the weights are preserved to make sure it's still true?  

Regarding timm and its interacations in our library: I believe it should be OK. The weights for the timm model are initialized in the model's init call before `post_init` and are marked with `is_hf_initialized=True`. I can add a test here to include timm to make sure it works as expected + protect against regressions.